skosprovider_sqlalchemy/utils.py

```diff
@@ -28,13 +31,27 @@ def import_provider(provider, conceptscheme, session):
     :param conceptscheme: A
         :class:`skosprovider_sqlalchemy.models.Conceptscheme` to import
         the provider into. This should be an empty scheme so that there are
-        no possible id clashes.
+        no possible id clashes. If no conceptscheme is provided, one will be
+        created.
     :param session:  A :class:`sqlalchemy.orm.session.Session`.
 
+    :return: The conceptscheme that holds the concepts and collections. Either
+        the same conceptscheme that was passed into the provider, or the one that
+        was created by this function.
+    :rtype: skosprovider_sqlalchemy.models.Conceptscheme
     '''
 
+    if not conceptscheme:
+        conceptscheme = ConceptSchemeModel()
```

Comment:
I would personally already fill in the uri here. Because the object is added to the session the line after.

After this, you're on the mercy of sqlalchemy when it gets flushed (unless you specifically turn autoflush off).
If it were to flush right now you'd crash because uri cannot be null.

I know in practice sqlalchemy only flushes when needed, so it wouldn't fail in this code. But it still feels safer to me to not have an incomplete object added to the session.